Sharadha taking over for Manvitha- Event Reschedule Button

myeeli opened the pull request:
# Description
<img width="550" height="320" alt="image" src="https://github.com/user-attachments/assets/f67a2ddd-6352-4c7e-ba98-33ced6b7918e" />

<img width="608" height="318" alt="image" src="https://github.com/user-attachments/assets/68a39c19-bc78-4ac3-a977-eda3860b927d" />

<img width="554" height="314" alt="image" src="https://github.com/user-attachments/assets/ca7d7fe4-e9b1-4e1f-ab1c-7c2aacd896f0" />
<img width="549" height="122" alt="image" src="https://github.com/user-attachments/assets/79ee99bf-f9b7-4bf9-9c97-e15f8e9d0dc2" />

## Related PRS (if any):
This frontend PR is related to the [#1856](https://github.com/OneCommunityGlobal/HGNRest/pull/1856) backend PR.

…

## Main changes explained:

…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to localhost:5173/communityportal/Activities
6. click on reschedule button

## Screenshots or videos of changes:


https://github.com/user-attachments/assets/2ea6bfe0-a4d0-4579-9677-a85012acc710


## Note:
Include the information the reviewers need to know.

VMD281 committed:
Your commit message

VMD281 committed:
reschedule-button

VMD281 committed:
reschedule button and datepicker component added succesfully

VMD281 committed:
Event details updated

VMD281 committed:
Tests failure fixed

VMD281 committed:
Update .eslintrc.json

VMD281 committed:
Resolved merge conflicts with main

VMD281 committed:
Fix: tests passing locally after changes

SammanBaidya committed:
fix: requested changes

SammanBaidya committed:
Merge

SammanBaidya committed:
ressolve dependency

myeeli committed:
Resolved node issues and merge conflicts, changed to module.css

myeeli committed:
Modified the ui of the modal and made it responsive for different screen sizes

myeeli committed:
Modified the reschedule button to activites list page and addedfeature for organiser to select multiple dates

myeeli committed:
Modified the reschedule button to activites list page and addedfeature for organiser to select multiple dates

myeeli committed:
API Call

myeeli committed:
Integrated API

myeeli committed:
Integrated polling in email

myeeli committed:
Fixing jwtcode issue

myeeli committed:
Implemented component to get the poll response

myeeli committed:
Resolved Merge Conflicts

myeeli committed:
Resolve Merge Conflicts

myeeli committed:
Resolve Merge Conflicts

myeeli committed:
Resolve Merge Conflicts

myeeli committed:
Resolve Merge Conflicts

myeeli committed:
Resolving lint issues

myeeli committed:
Resolving lint issues

myeeli committed:
Fixed linting issues

Anusha-Gali reviewed (changes requested):
Hi Manvitha,

I tried to review your PR locally but the UI is not how it is shown in the video you have kept and also - it is sending the email to 3 participants whose mails i have not given and also the form is not accessible. You can find my detailed review in the video below. Also note the page is not Dark Mode accessible

https://github.com/user-attachments/assets/e5d3abcb-3e86-49c1-9af4-dfec929dda95

VijayAnirudh reviewed (changes requested):
Hello! 
1) The font is not visible.
2) UI doesn't match with the video. (probably different browsers render UI differently.)
3) Options don't appear in-line, as shown in the video
4) I cannot send the mail! 
Please include an accurate link for obtaining env variables from the Google Cloud Console. The link you've provided explains how OAuth works.
Here's the correct link: https://developers.google.com/identity/protocols/oauth2/native-app

<img width="1908" height="910" alt="image" src="https://github.com/user-attachments/assets/d17fb061-0e65-4d1d-bc85-a9a95f43a66f" />
<img width="1186" height="742" alt="image" src="https://github.com/user-attachments/assets/12e22657-6058-4536-8b74-65e7975069f6" />
<img width="1235" height="350" alt="image" src="https://github.com/user-attachments/assets/c4e03f92-2f40-47a2-9b70-9632e49cdd0f" />
<img width="1050" height="812" alt="image" src="https://github.com/user-attachments/assets/a1a9d055-5270-4e79-8614-d8022f8e2790" />

myeeli commented:
> Hi Manvitha,
> 
> I tried to review your PR locally but the UI is not how it is shown in the video you have kept and also - it is sending the email to 3 participants whose mails i have not given and also the form is not accessible. You can find my detailed review in the video below. Also note the page is not Dark Mode accessible
> 
>  4276.mp4

the email id is hardcoded as by the time of development the event registration feature is not implemented completely to fetch all the emails ids registered for that event.

SharadhaKasiviswanathan committed:
Resolve development merge conflicts

SharadhaKasiviswanathan committed:
Fix dark mode for reschedule event views

SharadhaKasiviswanathan committed:
Fix all SonarCloud Quality Gate issues

- Remove unused imports in routes.jsx (hasPermission, RoleInfoCollections, EmailSubscribeForm, UnsubscribeForm, EmbedInteractiveMap, FollowUpEmailTemplate)
- Fix nested ternary in ActivityList.jsx by splitting into separate conditions
- Add PropTypes validation to RescheduleEvent and RescheduleModal components
- Fix negated condition (!confirmStep) by swapping branches
- Fix CSS contrast issues (disabled button colors, text on background)
- Use globalThis instead of window in ReschedulePoll.jsx

Copilot committed:
Merge development and resolve activity route conflicts

SharadhaKasiviswanathan reviewed (approved):
All SonarCloud Quality Gate issues fixed (Reliability 1.0/A). Darkmode feedback addressed in commit 54467b0. All requested changes from earlier reviews have been resolved.
